src/ray/util/process.cc

```diff
@@ -395,7 +395,7 @@ int Process::Wait() const {
   return status;
 }
 
-void Process::Kill() {
+void Process::Kill() const {
```

Comment:
Actually Boost.Process is what we tried before this. It turned out that Boost.Process was quite unstable and had bugs (and slowed down compilation dramatically too), and after struggling with it for over 1 month, we had to scrap it and roll our own solution, which works fine. Please don't revert back to it; it is very likely to cause problems again.

@fyrestone Regarding the `const`, this shouldn't be `const`. What was the reason for adding `const` here?

Reply:
> Actually Boost.Process is what we tried before this. It turned out that Boost.Process was quite unstable and had bugs (and slowed down compilation dramatically too), and after struggling with it for over 1 month, we had to scrap it and roll our own solution, which works fine. Please don't revert back to it; it is very likely to cause problems again.
> 
> @fyrestone Regarding the `const`, this shouldn't be `const`. What was the reason for adding `const` here?

Got it. Process could add some useful interface: WaitFor, WaitUntil.
Another suggestion, Process should forbidden copy construct and copy assign function, just move only will be better.
Hope you could add unit tests for Process, especially the exception tests.

Reply:
These are not trivial changes to make correctly. I don't think we should alter this specific class without compelling reasons.

The entire design of the API here was honestly quite deliberate. For example, I specifically allowed copying because the initial code used `pid_t` everywhere and this ensured semantic equivalence and minimized the changes of bugs being introduced. Requiring unique ownership would require a significant refactor of the rest of the codebase to accommodate it. If someone feels compelled to do it and can do it without introducing bugs, I suppose they can go ahead, but it is a significant effort and in the end there seems to be very little to gain.

Note that the primary intention for this class is to provide an easy-to-use interface that works correctly across all operating systems. While there _are_ certain features missing that could be added and potentially come in handy, the limited interface here was primarily designed to satisfy our actual needs without adding unnecessary maintenance burden. Process management is possibly the biggest source of pain when porting code across operating systems, and every change would have to be shown to work correctly on all 3 platforms with uniform semantics.

Overall, this is one of the few classes classes where it's not a good idea to change it just for the sake of elegance. If there is a functionality we actually *need*, then perhaps that warrants adding it. But as long as it is working correctly and other solutions can be reasonably implemented on top, it should probably be left alone.

Reply:
> Actually Boost.Process is what we tried before this. It turned out that Boost.Process was quite unstable and had bugs (and slowed down compilation dramatically too), and after struggling with it for over 1 month, we had to scrap it and roll our own solution, which works fine. Please don't revert back to it; it is very likely to cause problems again.
> 
> @fyrestone Regarding the `const`, this shouldn't be `const`. What was the reason for adding `const` here?

I have removed const from Process::Kill().